### ISSUE TITLE
Missing DUB build options

### ISSUE THREAD

**Issue by tsbockman** [kingsleyh/DLanguage#45]
The "Build with DUB" run configuration is rather useless right now, due to [missing options](http://code.dlang.org/docs/commandline#run).

The absolute must-have list, in my opinion, is:
* `--build=<value>` to select `release`, `debug`, `unittest`, etc.
* The ability to specify a sub-package to build, instead of locking it to the top level.
* `--compiler=<value>` to choose between `dmd`, `gdc`, and `ldc2`.
* `DUB build parameters` to cover options not added to the GUI form configuration yet.
* `Application parameters` to feed to the generated binary, if `Run after build` is set.

Also nice to have would be:
* `--arch=<value>` to select 32- or 64-bit.
* `--force` to rebuild.

All of that should be trivial to add (except perhaps sub-package support), so I might look into setting this up myself if you don't want/have too much else on your plate.

Aside from this issue (and the nasty DCD bug which you already fixed), I am quite pleased with the DLanguage plugin. I think it will be replacing Mono-D for me. The only *big* thing missing is debug support, but for me, at least, Mono-D effectively doesn't have debug support anyway since it crashes constantly...

Thanks for your efforts!

**Comment by kingsleyh** [comment]
Hi

I will add this to the next release - I'm thinking of creating 6 tabs in the config box as follows:

general - will contain a dropdown with command options
1. build
2. run
3. test
4. generate
5. dustmite

build - will contain all the possible build options (enabled only if build has been selected)
run - will contain all the possible run options (enabled only if run has been selected)
test - will contain all the possible test options (enabled only if test has been selected)
generate - same as the above
dustmite - same as the above

this way we can set up different configurations that do specific things 
What do you think?

**Comment by tsbockman** [comment]
That sounds good - just be sure to include an option manually enter the parameters, as a fall-back.

Otherwise you'll be under pressure to provide timely updates to the GUI every time some trivial change is made to DUB's options. (Currently I'm cheating and using the Cpp plugin's "run" command to accomplish this.)

**Comment by kingsleyh** [comment]
done and will be in next release

**Comment by kingsleyh** [comment]
I've pushed a new release with this functionality - I didn't do the sub path - although if you change the working directory maybe that will do it?
Also haven't added the pass to binary options - if you can give me a pointer on that I will add it 
give it a spin and let me know if it meets your needs

**Comment by tsbockman** [comment]
> I've pushed a new release with this functionality
> give it a spin and let me know if it meets your needs

Excellent. I took a quick look, and the "Build with DUB" configuration is actually useful now.

I'll probably be back to request some minor tweaks after I've tested it more thoroughly.

> I didn't do the sub path - although if you change the working directory maybe that will do it?

Nope. That was my first guess also, but it turns out it actually works like this:
    `dub build [package name]:[sub-package name]`
For now you could just add an additional text box between "Module" and "Dub options" to specify the sub-package, although eventually it would be nice to have the IDE detect all the sub-packages within each module and present the user with a list.

> Also haven't added the pass to binary options - if you can give me a pointer on that I will add it

Per the [DUB CLI docs](http://code.dlang.org/docs/commandline):
    `dub [--version] [<command>] [<options...>] [-- [<application arguments...>]]`
So basically, add a double dash `-- ` after all the DUB arguments, and then paste in the contents of the pre-existing "Parameters" box afterwards.

**Comment by kingsleyh** [comment]
Hi - great thanks for the feedback. Raising feature requests/issues is very helpful for me to improve the plugin - so feel free :) 

I will add the text box as you suggest for sub packages and the pass to binary options

**Comment by kingsleyh** [comment]
do you have an example project that has sub-packages - I haven't used that feature before and I wanted to see how it works. How would I find the sub packages programmatically - would I have to parse the dub.json/sdl to get them ? Or is there a better way

**Comment by tsbockman** [comment]
@kingsleyh 
> do you have an example project that has sub-packages

https://code.dlang.org/packages/checkedint
